core/src/main/java/org/lflang/generator/c/CGenerator.java

```diff
@@ -575,18 +522,47 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
             this::reportCommandErrors,
             context.getMode());
         context.finish(GeneratorResult.Status.COMPILED, null);
+      } else if (dockerBuild.enabled()) {
+        buildUsingDocker();
+      } else {
+        var cleanCode = code.removeLines("#line");
+        var cCompiler = new CCompiler(targetConfig, fileConfig, messageReporter, cppMode);
+        try {
+          if (!cCompiler.runCCompiler(this, context)) {
+            // If compilation failed, remove any bin files that may have been created.
+            CUtil.deleteBinFiles(fileConfig);
+            context.unsuccessfulFinish();
+          } else {
+            context.finish(GeneratorResult.Status.COMPILED, null);
+          }
+          cleanCode.writeToFile(targetFile);
+        } catch (IOException e) {
+          messageReporter.nowhere().warning("Generated code may still contain line directives.");
+        }
```

Comment:
I'm not sure I'm comfortable handling all errors that occur in the above `try` block using this error message. What if the IOException comes from `runCCompiler` and not `cleanCode.writeToFile(...)`? It's possible that `runCCompiler` and `cleanCode.writeToFile` should go in separate `try` blocks so that we can handle the possible errors in a more specific/granular way.

Previously, the behavior was to show the user a stack trace, I think. The rule of thumb that I am applying here is that it is better to fail ungracefully and show the user an ugly stack trace than to provide a message that may be inaccurate and show the user nothing else.

Reply:
Previously, a `sneakyThrow` was being used, which I think is worse than printing an at least somewhat helpful error message...

Reply:
But above I described an easy way to make the error message more helpful: use a separate `try` block for `runCCompiler`. Even if you think this is better than printing a stack trace, at least we can agree that if the error came from `runCCompiler`, then the error message should be different.

Reply:
OK, yeah, we can do more `try` `catches`. Will push a fix.

Reply:
Check again? 8b5e98ac3da1706e92a78f939a36edc30b9d43f4